src/PermissionsContext.tsx

```diff
@@ -23,25 +23,32 @@ const initialENPermissionStatus: ENPermissionStatus = [
   "DISABLED",
 ]
 
-const toAuthorizationEnablementStatus = (
-  enPermissionStatus: ENPermissionStatus,
-): ENAuthorizationEnablementStatus => {
-  const isENAuthorized = enPermissionStatus[0] === "AUTHORIZED"
-  const isENEnabled = enPermissionStatus[1] === "ENABLED"
-  const isENAuthorizedAndEnabled = isENAuthorized && isENEnabled
-
-  return {
-    authorized: isENAuthorized,
-    enabled: isENEnabled,
-    authorizedAndEnabled: isENAuthorizedAndEnabled,
+const toENStatus = (enPermissionStatus: ENPermissionStatus): ENStatus => {
+  const isAuthorized = enPermissionStatus[0] === "AUTHORIZED"
+  const isEnabled = enPermissionStatus[1] === "ENABLED"
+
+  if (!isAuthorized && !isEnabled) {
+    return ENStatus.UNAUTHORIZED_DISABLED
+  }
+
+  if (isAuthorized && !isEnabled) {
+    return ENStatus.AUTHORIZED_DISABLED
+  }
+
+  if (isAuthorized && isEnabled) {
+    return ENStatus.AUTHORIZED_ENABLED
   }
+
+  return ENStatus.UNAUTHORIZED_DISABLED
 }
-export type ENAuthorizationEnablementStatus = {
-  authorized: boolean
-  enabled: boolean
-  authorizedAndEnabled: boolean
+
+export enum ENStatus {
```

Comment:
Yeah! this seems like a better fit for our use case.

an alternative would be a union type, which would just change the look n feel and not the type safety:
```typescript
export type ENStatus = "AuthorizedEnabled" | "AuthorizedDisabled" | "UnauthorizedDisabled"
```